### ISSUE TITLE
Expose AMI Family as node label

### ISSUE THREAD

**Issue by runningman84** [aws/karpenter-provider-aws#3074]
### Tell us about your request

Right now it is difficult to find out if a given instance runs bottlerocket or AL2. 

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

For example In bottlerocket NFS mount does not work... it would be great if we could schedule this nfs stuff to AL2 instances...

### Are you currently working around this issue?

right now we run the whole cluster in AL2

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
``

**Comment by FernandoMiguel** [comment]
wouldnt those be two different `AWSNodeTemplate`?
so you can place different selectors, no ?

**Comment by runningman84** [comment]
we might have n nodes templates but only two different os types....

**Comment by jonathan-innis** [comment]
You could do `nodeAffinitiy` or `nodeSelector` specifying all the provisioners that use the correct AMIFamily that you are looking for, like

```
nodeSelector:
  karpenter.sh/provisioner-name: default
```

The prospect of having AWSNodeTemplate generated requirements is interesting but it does change the way we have delineated what affects scheduling and what affects launching. Currently, the AWSNodeTemplate has little affect on scheduling beyond the zones that are supported based on the passed `subnetSelector`.

**Comment by runningman84** [comment]
Yes that is possible but developers might not know all these details. They might just want to specify the underlying ami family.

**Comment by jonathan-innis** [comment]
I agree that this could be a useful for users that don't want to have to pin to specific provisioners.

We're in a bit of flux right now due to a lot of other changes that are happening in the project so we're trying to reduce API changes right now until we get past https://github.com/aws/karpenter/pull/2964 and https://github.com/aws/karpenter/pull/2944. With that in mind, we'll keep tracking this issue but expanding the label selector set isn't something that's going to get added immediately.

It seems like you have a reasonable workaround for the time being.

**Comment by jfoechsler** [comment]
Hello. Any updates to having such an AMI label to use as node selector?
